ezyrb/regular_grid.py

```diff
@@ -0,0 +1,131 @@
+"""
+Module for higher order interpolation on regular grids
+"""
+import numpy as np
+from scipy.interpolate import RegularGridInterpolator
+from .approximation import Approximation
+
+
+class RegularGrid(Approximation):
+    """
+    Multidimensional interpolator on regular grids.
+    See scipy's RegularGridInterpolator
+
+    """
+
+    def __init__(self):
+        self.interpolator = None
+        self.dim = None
+        self.n_modes = 0
+        self.mode_nr = None
+
+    def get_grid_axes(self, pts_scrmbld, vals_scrmbld):
+        """
+        calculates the grid axes from a meshed grid. The grid axes are given as
+        a tuple of ndarray of float, with shapes (m1, ), …, (mn, )
+        The values are ordered so they fit on a mesh generated with
+        numpy.meshgrid(ax1, ax2, ..., axn, indexing="ij")
+
+        """
+        # be aware of floating point precision in points!
+        grid_axes = []
+        iN = []  # index in dimension N
+        nN = []  # size of dimension N
+        dim = pts_scrmbld.shape[1]
+        for i in range(dim):
+            xn, unique_inverse_n = np.unique(pts_scrmbld[:, i],
+                                             return_inverse=True)
+            grid_axes.append(xn)
+            nN.append(len(xn))
+            iN.append(unique_inverse_n)
+
+        if np.prod(nN) != len(vals_scrmbld):
+            raise ValueError("points and values are not on a regular grid")
```

Comment:
I'm not sure this error is the best we can do. As a user I'd like to get some more info on the cause of the error. I propose something like:
```suggestion
            raise ValueError("Points and values are not laid out on a regular grid, the size of vals_scrmbld is not equal to ...")
```
and replacing the dots with something meaningful.